lua/bufferline/options.lua

```diff
@@ -80,10 +82,10 @@ function options.exclude_name()
   return get('exclude_name', {})
 end
 
+--- @param icon_option bufferline.options.icons
 --- @return boolean enabled
-function options.file_icons()
-  local enabled = options.icons()
-  return enabled == true or enabled == 'both' or enabled == 'buffer_number_with_icon'
+function options.file_icons(icon_option)
```

Comment:
Perf improvement. Passing in the `options.icons()` return value allows it to be used for multiple tests, e.g.:

```lua
local icons = options.icons()
local foo = options.file_icons(icons)
local bar = options.index_buffers(icons)
```

Before, each call to `file_icons`, etc. had to call `options.icons()` (which in turn called `get('icons', true)`) in order to return a value. Now it's just a set of test cases.